src/main/java/io/github/fablabsmc/fablabs/api/fluidvolume/v1/FluidHelper.java

```diff
@@ -0,0 +1,83 @@
+package io.github.fablabsmc.fablabs.api.fluidvolume.v1;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.containers.BlockFluidContainerProvider;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.containers.EntityFluidContainerProvider;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.world.WorldFluidCallback;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.containers.ItemFluidContainerProvider;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.ImmutableFluidVolume;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.MultiFluidContainer;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.api.FluidContainer;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Box;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.World;
+
+public final class FluidHelper {
+	/**
+	 * attempt to get a fluid container from the world from a specific block (or entities) for a given face.
+	 *
+	 * @param world the world
+	 * @param pos   the position
+	 * @param side  the side of the block/entity to access
+	 * @return ImmutableFluidVolume#EMPTY if no fluid container exists for the given side
+	 */
+	public static FluidContainer getContainer(World world, BlockPos pos, Direction side) {
+		BlockState state = world.getBlockState(pos);
+		Block block = state.getBlock();
+
+		FluidContainer container = ImmutableFluidVolume.EMPTY;
+
+		if (block instanceof BlockFluidContainerProvider) {
+			container = ((BlockFluidContainerProvider) block).getContainer(world, state, pos, side);
+		} else {
+			FluidContainer worldContainer = WorldFluidCallback.EVENT.invoker().getContainer(world, pos);
+
+			if (worldContainer != null) {
+				container = new MultiFluidContainer(container, worldContainer);
+			}
+		}
+
+		List<FluidContainer> containers = new ArrayList<>();
+
+		for (Entity entity : world.getEntities((Entity) null, new Box(pos), e -> e instanceof EntityFluidContainerProvider)) {
+			containers.add(((EntityFluidContainerProvider) entity).getContainer(side));
+		}
+
+		if (!containers.isEmpty()) {
+			containers.add(container);
```

Comment:
Maybe check if this is null

Reply:
how can con tater be null
![image](https://user-images.githubusercontent.com/28693059/82755916-66994080-9d9c-11ea-956a-2489f21ce05e.png)

Reply:
the `container` being `null`